src/row.rs

```diff
@@ -80,6 +80,11 @@ impl Row {
     pub fn cell_iter(&self) -> Iter<Cell> {
         self.cells.iter()
     }
+
+    /// Return the cells of this row.
+    pub fn cells(&self) -> Vec<Cell> {
+        self.cells.clone()
+    }
```

Comment:
This is a no-go. You're copying the cells even though you only need to read them for comparison reasons.

Please make this return a `&[Cell]`